packages/react/src/menu/positioner/MenuPositioner.test.tsx

```diff
@@ -315,29 +322,31 @@ describe('<Menu.Positioner />', () => {
         </Menu.Root>,
       );
 
-      const trigger = getByRole('button', { name: 'Toggle' });
+      const trigger = screen.getByRole('button', { name: 'Toggle' });
 
-      expect(queryByRole('menu', { hidden: true })).not.to.equal(null);
-      expect(queryByRole('menu', { hidden: true })).toBeInaccessible();
+      expect(screen.queryByRole('menu', { hidden: true })).not.to.equal(null);
+      expect(screen.queryByRole('menu', { hidden: true })).toBeInaccessible();
 
-      await user.click(trigger);
-      await flushMicrotasks();
+      await user.click(trigger, { delay: 20 });
       await waitFor(() => {
-        expect(queryByRole('menu', { hidden: false })).not.to.equal(null);
+        expect(screen.queryByRole('menu', { hidden: false })).not.to.equal(null);
       });
-      expect(queryByRole('menu', { hidden: false })).not.toBeInaccessible();
+      expect(screen.queryByRole('menu', { hidden: false })).not.toBeInaccessible();
 
-      await user.click(trigger);
+      await user.click(trigger, { delay: 20 });
```

Comment:
This is an interesting case. I spend quite some time debugging why everything works perfectly fine and yet tests fail when clicking on the trigger twice. It turns out that test tools (both RTL and Playwright) generate mouse events so quickly one after the other (mousedown + click), that our code doesn't have a chance to run in between. In real-life usage, the handler for `mousedown` would always fire before the `click` event, setting the `ignoreClickRef` value (and preventing the menu from closing right after it was opened). In tests, `click` was fired before the ref was set and the menu closed immediately.